### PR TITLE
Allow configuring or disabling max cookie size check

### PR DESCRIPTION
**Pull request by davidism**
Continue #780 

* use arg instead of const for `dump_cookie` `max_size`
* add `max_cookie_size` attr to `BaseResponse`
* add changelog

### CODE REVIEW

File: werkzeug/wrappers.py [hunk 1]
```diff
@@ -1072,15 +1083,11 @@ def set_cookie(self, key, value='', max_age=None, expires=None,
                          extension to the cookie standard and probably not
                          supported by all browsers.
         """
-        self.headers.add('Set-Cookie', dump_cookie(key,
-                                                   value=value,
```
Comment (untitaker) [inline werkzeug/wrappers.py]: Can we keep the vertical alignment of all keys? IMO easier to read. Of course the linebreak before the first one is an improvement.

**Comment by untitaker** [comment]
LGTM, might need a testcase for custom values though

**Comment by davidism** [comment]
The error is slightly confusing because you can pass 4093 bytes as the value and still get the error, since the options are added to the payload. I don't think we need to change it, just pointing it out.

**Comment by untitaker** [comment]
I think we should elaborate the error message because of this.

In Flask it might make sense to only enable this limit in developer mode, and
rather try not to crash in production.

**Comment by davidism** [comment]
Maybe this should be a warning? Could use a filter to turn it into an error in Flask debug mode.

**Comment by davidism** [comment]
How about detecting the before and after version, then warning "Cookie too large: value was {value_size} bytes but header required {header_size - value_size} extra bytes. The final size was {header_size} but the limit is {max} bytes. Browsers may silently ignore cookies larger than this."

**Comment by untitaker** [comment]
Sounds good

On April 18, 2017 6:24:54 PM GMT+02:00, David Lord <notifications@github.com> wrote:
>How about detecting the before and after version, then warning "Cookie
>too large: value was {value_size} bytes but header required
>{header_size - value_size} extra bytes. The final size was
>{header_size} but the limit is {max} bytes. Browsers may silently
>ignore cookies larger than this."
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/pallets/werkzeug/pull/1109#issuecomment-294899723

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by Jaza** [comment]
Thanks @davidism for making this configurable, there should be need to override the limit of 4k per cookie in most cases, but yes, some people might want to change it.

**Pull request merged**